src/core/net/ip6_address.hpp

```diff
@@ -324,6 +324,17 @@ class Address : public otIp6Address, public Equatable<Address>, public Clearable
         kGlobalScope         = 14, ///< Global scope
     };
 
+    /**
+     * This enumeration defines IPv6 address type filter.
+     *
+     */
+    enum TypeFilter : uint8_t
+    {
+        kTypeAny,           ///< Accept any IPv6 address (unicast or multicast).
+        kTypeUnicastOnly,   ///< Accept unicast IPv6 addresses only.
+        kTypeMulticastOnly, ///< Accept multicast IPv6 addresses only.
```

Comment:
```suggestion
        kTypeMulticastOnly, ///< Accept multicast IPv6 addresses only.
        kTypeMulticastLargerThanRealmLocal, ///< Accept multicast IPv6 addresses with scope larger than Realm Local.
```
kTypeMulticastLargerThanRealmLocal can be very useful for MLR features.

Reply:
Will add it later

Reply:
Changed in new pushed commit.